pkg/k8sapi/discovery.go

```diff
@@ -185,6 +185,11 @@ func (d *Controller) K8SGetLocalPodIPs() ([]*K8SPodInfo, error) {
 	return localPods, nil
 }
 
+func (d *Controller) WaitUntilReady() {
+	stopCh := make(chan struct{})
+	cache.WaitForCacheSync(stopCh, d.controller.informer.HasSynced)
```

Comment:
This sounds like something we should do, but the go docs say `WaitForNamedCacheSync` should be preferred. Could you change to used that function instead? Also, I'll try to test this some before it gets merged.